feat(next): better tsconfig (#11859)
withastro/astro: .changeset/old-zebras-teach.md

---
'astro': major
---

> Updates `astro sync` behavior

Reviewer: It seems to me like maybe the *main* idea is that Astro no longer relies on `env.d.ts` for configuring TS, but rather moves functionality to `tsconfig.json`? And, as a side effect, `astro sync` behaviour is changed because that file is no longer the recommended place for configuring that stuff?

Would that be a more helpful framing of what's going on here for people?

Author: Yes exactly, Although it's still possible to use `env.d.ts`.

```suggestion
Removes `astro sync` update of `src/env.d.ts`
```
Something like that? Pretty bad I'd say :/

Reviewer: Maybe something like?

```suggestion
Changes where TypeScript `include` and `exclude` configuration is set, and updates `astro check` behavior to no longer rely on `src/env.d.ts` 
```

Author: ```suggestion
Changes the default `tsconfig.json` with better defaults, and makes `src/env.d.ts`  optional
```

wdyt?

Reviewer: That works!

> `astro sync` will no longer create nor update `src/env.dts`. You can keep using it but we recommend you update your root `tsconfig.json` instead:

Reviewer: ```suggestion
Astro's TypeScript configuration has been updated to [[read `include` and `exclude` properties from]](however you're now "using" this config file) `tsconfig.json` by default. This means that `src/env.d.ts` is only necessary if you have added custom ... (whatever we said last time?).

Additionally, running `astro sync` no longer creates, nor updates, `src/env.d.ts` as it is not required for type-checking standard Astro projects.

To update your project to Astro's recommended TypeScript settings, please add the following `include` and `exclude` properties to `tsconfig.json`:
```
just some blah blah up here  with very not specific wording!! Can you work with this?

---
'astro': major
---

Updates `astro sync` behavior

> Astro examples default `tsconfig.json` has been updated to include generated types and exclude your build output. This means that `src/env.d.ts` is only necessary if you have added custom or if you're not using a `tsconfig.json` file.

Reviewer: ```suggestion
Astro's default `tsconfig.json` in starter examples has been updated to include generated types and exclude your build output. This means that `src/env.d.ts` is only necessary if you have added custom configurations or if you're not using a `tsconfig.json` file.
```
We need a noun after "custom..." custom settings? custom configurations? custom types?
